scipy/signal/signaltools.py

```diff
@@ -458,6 +470,15 @@ def convolve(in1, in2, mode='full'):
     if volume.ndim == kernel.ndim == 0:
         return volume * kernel
 
+    # see whether the fourier transform convolution method or the direct
+    # convolution method is faster (discussed in scikit-image PR #1792)
+    direct_time = np.prod(volume.shape + kernel.shape)
+    fft_time = np.sum([n*np.log(n) for n in volume.shape + kernel.shape])
+    time_ratio = 40.032 * fft_time / direct_time
```

Comment:
Also, this may be a premature micro-optimization, but if instead of `time_ratio` you computed
```python
time_diff = 40.032 * fft_time - direct_time
```
you could later use `time_diff < 0` as your condition, replacing an always expensive division by a cheap subtraction, and without hurting readability much, if at all.